flink-table/flink-table-common/src/main/java/org/apache/flink/table/catalog/Catalog.java

```diff
@@ -43,6 +45,16 @@
 @PublicEvolving
 public interface Catalog {
 
+	/**
+	 * Get an optional {@link TableFactory} instance that's responsible to generate source/sink for tables
```

Comment:
```suggestion
	 * Get an optional {@link TableFactory} instance that's responsible for generating source/sink for tables
```

Also, can you mention in javadoc when and where this is supposed to be called? Audience may get confused on how this cooperates with table discovery service.